internal/config/config.go

```diff
@@ -134,18 +137,37 @@ func (c *Config) NewFabricClientForSDK(d *schema.ResourceData) *fabricv4.APIClie
 // Shim for Fabric tests.
 // Deprecated: when the acceptance package starts to contain API clients for testing/cleanup this will move with them
 func (c *Config) NewFabricClientForTesting() *fabricv4.APIClient {
-	client := c.newFabricClient()
+	client := c.newFabricClient(context.WithValue(context.Background(), ctxKey("fabricSDKv2"), "no_oauth_reload"))
 
 	client.GetConfig().UserAgent = fmt.Sprintf("tf-acceptance-tests %v", client.GetConfig().UserAgent)
 
 	return client
 }
 
+func (c *Config) NewFabricClientForFramework(ctx context.Context, meta tfsdk.Config) *fabricv4.APIClient {
+	client := c.newFabricClient(ctx)
+
+	baseUserAgent := c.tfFrameworkUserAgent(client.GetConfig().UserAgent)
+	client.GetConfig().UserAgent = generateFwModuleUserAgentString(ctx, meta, baseUserAgent)
+
+	return client
+}
+
 // newFabricClient returns the base fabricv4 client that is then used for either the sdkv2 or framework
 // implementations of the Terraform Provider with exported Methods
-func (c *Config) newFabricClient() *fabricv4.APIClient {
+func (c *Config) newFabricClient(ctx context.Context) *fabricv4.APIClient {
 	//nolint:staticcheck // We should move to subsystem loggers, but that is a much bigger change
 	transport := logging.NewTransport("Equinix Fabric (fabricv4)", c.authClient.Transport)
+	if v := ctx.Value(ctxKey("fabricSDKv2")); v == nil {
```

Comment:
IMO it'd be better to have two functions, one that takes a context and includes authClient creation, and the old one that didn't take a context.

I'm curious what is the benefit of choosing not to create a new authClient, though?